test/new_order.spec.ts

```diff
@@ -515,10 +515,135 @@ describe('NewOrder', function () {
+          amount: 0.1,
+          uniqId: 'testing-1111',
+          userAddr: userAddr,
+          protocol: Protocol.RFQV1,
+        },
+      })
+      expect(signedOrderResp).is.not.null
+      // verify data object
+      const order = signedOrderResp.order
+      console.log(order)
+      expect(order).is.not.null
+      expect(order.protocol).eq(Protocol.RFQV1)
+      expect(order.quoteId).eq('1--echo-testing-8888')
+      expect(order.makerAddress).eq(mmproxy.address.toLowerCase())
+      expect(order.makerAssetAmount).eq('100000')
+      expect(order.makerAssetAddress).eq('0xdac17f958d2ee523a2206206994597c13d831ec7')
+      expect(order.makerAssetData).eq(
+        '0xf47261b0000000000000000000000000dac17f958d2ee523a2206206994597c13d831ec7'
+      )
+      expect(order.takerAddress).eq(userAddr)
+      expect(order.takerAssetAmount).eq('100000000000000000')
+      expect(order.takerAssetAddress).eq(WETH[chainId].toLowerCase())
+      expect(order.takerAssetData).eq(
+        `0xf47261b0000000000000000000000000${WETH[chainId].toLowerCase().slice(2)}`
+      )
+      expect(order.senderAddress).eq('0xd489f1684cf5e78d933e254bd7ac8a9a6a70d491')
+      expect(order.feeRecipientAddress).eq('0xb9e29984fe50602e7a619662ebed4f90d93824c7')
+      expect(order.exchangeAddress).eq('0x30589010550762d2f0d06f650d8e8b6ade6dbf4b')
+      // The following fields are to be compatible `Order` struct.
+      expect(order.makerFee).eq('0')
+      expect(order.takerFee).eq('0')
+      // verify signature type
+      const sigBytes = utils.arrayify(signedOrderResp.order.makerWalletSignature)
+      expect(sigBytes.length).eq(66)
+      expect(sigBytes[65]).eq(SignatureType.WalletBytes32)
+      // verify random values
+      expect(signedOrderResp.order.salt.length > 0).is.true
+      expect(Number(signedOrderResp.order.expirationTimeSeconds) > 0).is.true
+      const rfqAddr = updaterStack['markerMakerConfigUpdater'].cacheResult.addressBookV5.RFQ
+      const orderSignDigest = getOrderSignDigest(toRFQOrder(signedOrderResp.order), 1, rfqAddr)
+      const r = utils.hexlify(sigBytes.slice(0, 32))
+      const s = utils.hexlify(sigBytes.slice(32, 64))
+      const v = utils.hexlify(sigBytes.slice(64, 65))
+      console.log(`r: ${r}`)
+      console.log(`s: ${s}`)
+      console.log(`v: ${v}`)
+      const recovered = utils.verifyMessage(utils.arrayify(orderSignDigest), {
+        v: parseInt(v),
+        r: r,
+        s: s,
+      })
+      expect(recovered.toLowerCase()).eq(mmpSigner.address.toLowerCase())
+      console.log(`recovered.toLowerCase(): ${recovered.toLowerCase()}`)
+      console.log(`mmpSigner.address.toLowerCase(): ${mmpSigner.address.toLowerCase()}`)
+    }).timeout(360000)
     it('should signed rfqv1 order by EOA', async function () {
       const userAddr = Wallet.createRandom().address.toLowerCase()
       const signedOrderResp = await newOrder({
-        signatureType: SignatureType.Wallet,
+        walletType: WalletType.MMP_VERSOIN_4,
```

Comment:
I might be mistaken. Should this be EOA wallet type?

Reply:
Yeah, you're right. This should be an EOA test case. BTW, if the signer address is the same as the address of `mmProxyContractAddress`, the mmsk itself will automatically use EOA for signing, otherwise it will sign for the wallet type. That's why the test is passed.